src/libraries/System.Private.CoreLib/src/System/Random.Xoshiro256StarStarImpl.cs

```diff
@@ -45,6 +45,10 @@ public unsafe Xoshiro256StarStarImpl()
                 while ((_s0 | _s1 | _s2 | _s3) == 0); // at least one value must be non-zero
             }
 
+            /// <summary>Produces a value in the range [0, uint.MaxValue].</summary>
+            [MethodImpl(MethodImplOptions.AggressiveInlining)] // small-ish hot path used by very few call sites
```

Comment:
Surely this one liner inlines itself

Reply:
I did it for consistency across implementations, so the methods in both cases have the same visibility, same attributes, etc.